### PR TITLE
feat(nz): Phase 1 — 6 kennels, 8 sources, first NZ coverage

### PR DESCRIPTION
**Pull request by johnrclem**
## Summary

HashTracks had zero New Zealand coverage. This Phase 1 PR seeds 4 metros (Auckland, Wellington, Christchurch, Hamilton/Waikato) and 6 kennels with 8 sources. All sources are live-verified against production URLs.

- 2 new HTML adapters (Miteri-theme shared parser for GCH3 + CHH3, Auckland Hussies Excel-export)
- 8 sources backed by 5 adapter types: Miteri-HTML × 2, STATIC × 4, GOOGLE_SHEETS × 1, freeform HTML × 1
- New shared utility `bumpYearIfBefore` for chronologically-sorted run lists (with leap-year clamp)
- New `Mon-DD` date format support in GoogleSheetsAdapter (Hibiscus uses this layout)
- Research artifact at [docs/kennel-research/new-zealand-research.md](docs/kennel-research/new-zealand-research.md) covers the broader 28-kennel NZ scene

## Live verification (today, 2026-05-15)

| Source | Adapter | Events | Date range |
|---|---|---:|---|
| Garden City H3 | Miteri (SiteOrigin) | 5 | 2026-05-22 → 2026-06-16 |
| Christchurch H3 | Miteri (Gutenberg) | 0* | n/a |
| Hibiscus H3 | Google Sheets (pubhtml) | 1 | 2026-05-18 |
| Auckland Hussies | Excel-export HTML | 9 | 2026-04-24 → 2026-06-16 |

\* CHH3's Gutenberg table is currently empty — STATIC weekly Monday baseline covers it until the kennel populates the table.

STATIC_SCHEDULE sources (Tokoroa dual seasonal + T3H3 + CHH3 baseline) generate occurrences from RRULE — covered by existing static-schedule tests.

## Edge case caught + fixed in adversarial review

`bumpYearIfBefore` used string-concat year arithmetic; a Feb 29 input bumped to a non-leap year produced `2025-02-29` which `new Date()` silently rolls to March 1. Now clamps to Feb 28 with unit-test coverage.

## Out of scope (subsequent PRs)

- **Phase 1.5** — sporty.co.nz spike (residential proxy / browserRender for the 403-blocked Capital, Mooloo, Geriatrix kennels)
- **Phase 2** — sporty unlocks
- **Phase 3** — STATIC bulk for ~18 remaining active NZ kennels
- **Phase 4** — FB Page audit for FACEBOOK_HOSTED_EVENTS upgrades

## Test plan

- [x] `npx tsc --noEmit` clean
- [x] `npm run lint` clean (no new warnings)
- [x] `npm test` — all 6464 tests pass (252 files)
- [x] Live verification via `npx tsx scripts/verify-nz-adapters.ts`
- [ ] Post-merge: confirm the new kennels appear on the hareline at `/hareline?region=new-zealand`
- [ ] Post-merge: confirm regional cards/maps render with `Pacific/Auckland` timezone

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by johnrclem** [comment]
## Codex adversarial review — response

**Finding 1 (Auckland Hussies, HIGH): DISMISSED**

Codex misread the research doc. The 403 it cited was a transient apex-page issue (`aucklandhussies.co.nz/`) during the initial discovery agent's pass. Our adapter targets `/Run%20List.html` instead, which the research doc explicitly notes returns 200 (and which we live-verified during the original PR by parsing 9 events). Just re-checked:

```
HTTP 200 - 56379b   (Run List page)
HTTP 200 - 17970b   (apex — even the original 403 has cleared)
```

Live verification post-fix still emits 9 events 2026-04-24 → 2026-06-16. No change needed.

**Finding 2 (Hibiscus runNumber, HIGH): ACCEPTED — fixed in 3a6c323a**

Codex is right — `runNumber` is in the fingerprint hash ([fingerprint.ts:31](src/pipeline/fingerprint.ts#L31)), and the Hibiscus sheet's column 0 is a sequential row counter, not a hash run number. Whenever the kennel adds/removes rows above, all downstream runNumbers shift, re-fingerprinting unchanged events.

Made `GoogleSheetsConfig.columns.runNumber` optional. When the column isn't configured, `resolveKennelTagFromSheetRow` emits `runNumber: undefined` and row validity is gated entirely by the date column (already filtered upstream). Dropped runNumber from the Hibiscus config. New unit test covers the optional-column path.

**Finding 3 (Christchurch homepage scraper, MEDIUM): ACCEPTED — dropped in 3a6c323a**

Codex is right that the homepage Miteri table is currently empty and CHH3's actual special events live on `/events/` as plain `<ul>/<li>` (verified — Blackball 13-15 March 2026, Winter Camp 10-12 July 2026). The Miteri adapter is a homepage-table parser, so keeping the source enabled with no real ingestion path creates false coverage.

Dropped the CHH3 HTML source row, the registry URL pattern, and the verify probe. STATIC weekly Monday baseline still covers the recurring cadence. The `/events/` parser is scoped for Phase 2.

Phase 1 now ships **7 sources** (was 8). Tests: 6467 passing. Live verification of GCH3 (5 events), Auckland Hussies (9 events), and Hibiscus (1 event) all green.

File: src/adapters/utils.ts [hunk 1]
```diff
@@ -808,6 +808,41 @@ export function stripPlaceholder(value: string | undefined | null): string | und
...
+  if (!prevDate || date > prevDate) return date;
+  const [origY, m, d] = date.split("-").map(Number);
+  let year = origY;
+  let result = date;
+  // The (m, d) anchor stays fixed; we only walk the year forward. This
+  // preserves Feb 29 across leap-year bumps — the final emission re-checks
+  // leap-year validity and clamps Feb 29 → Feb 28 only when the landing
+  // year is non-leap (otherwise downstream `new Date()` silently rolls to
+  // March 1).
+  while (result <= prevDate) {
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/utils.ts]: Fixed in d69dac0a. Switched `<=` → strict `<` in `bumpYearIfBefore` and updated the test to assert same-day rows return unchanged. Reasoning: in practice, same-day duplicates are typos/duplicate rows, not year rolls; the safer default is to leave them in the current year. Genuine same-day year rolls are vanishingly rare and would need explicit metadata.

File: src/adapters/html-scraper/miteri-hareline.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    opts.referenceDate,
+    { forwardDate: false },
+  );
+  if (!parsed) return null;
+  const date = bumpYearIfBefore(parsed, opts.prevDate);
+
+  const runClean = cleanCellText(row.runText);
+  const runNumber =
+    extractHashRunNumber(row.runText)
+    ?? (runClean ? Number.parseInt(runClean, 10) || undefined : undefined);
```
[the remark this answers is not in the document]
Comment (johnrclem) [reply src/adapters/html-scraper/miteri-hareline.ts]: Fixed in d69dac0a. Replaced `Number.parseInt(runClean, 10) || undefined` with an explicit `Number.isFinite()` check so a legitimate run number 0 is preserved instead of being silently coerced to undefined.
